avcodec.cpp

```diff
@@ -406,6 +409,33 @@ static int avcodec_decoder_copy_frame(const avcodec_decoder d, opencv_mat mat, A
             SWS_BILINEAR, // Specify the scaling algorithm; you can choose another according to your needs
             NULL, NULL, NULL);
 
+        // Configure colorspace conversion
+        int colorspace;
+        switch (frame->colorspace)
+        {
+        case AVCOL_PRI_BT2020:
+            colorspace = SWS_CS_BT2020;
+            break;
+        case AVCOL_PRI_BT709:
+            colorspace = SWS_CS_ITU709;
+            break;
+        case AVCOL_PRI_BT470BG:
+        case AVCOL_PRI_SMPTE170M:
+            colorspace = SWS_CS_ITU601;
+            break;
+        default:
+            colorspace = SWS_CS_ITU709;
+            break;
+        }
+
+        int srcRange;
+        if (frame->color_range != AVCOL_RANGE_UNSPECIFIED) {
+            srcRange = frame->color_range;
+        } else {
+            srcRange = AVCOL_RANGE_MPEG;
```

Comment:
`AVCOL_RANGE_MPEG` is 1, which sws will take to mean full-range, which isn't what you want here.